kong/plugins/request-termination/handler.lua

```diff
@@ -59,21 +67,29 @@ function RequestTerminationHandler:access(conf)
   local content = conf.body
 
   if content then
-    local ctx = ngx.ctx
-    if ctx.delay_response and not ctx.delayed_response then
-      ctx.delayed_response = {
+    local shared_ctx = kong.ctx.shared
```

Comment:
The whole delayed_response logic (and flush function) can be replaced by `kong.response.exit`, which takes care of it. It is possible to just do

```lua
if content then
  return kong.response.exit(status, content, {
    ["Content-Type"] = conf.content_type
  })
end
```

instead of this whole block, also removing the `flush` function above.